temporalio/worker/_workflow_instance.py

```diff
@@ -1954,10 +2011,75 @@ def _enhanced_stack_trace(self) -> temporalio.api.sdk.v1.EnhancedStackTrace:
             )
             return est
 
+    def _temporal_workflow_metadata(self) -> temporalio.api.sdk.v1.WorkflowMetadata:
+        query_definitions = [
+            temporalio.api.sdk.v1.WorkflowInteractionDefinition(
+                name=qd.name if qd.name is not None else "",
+                description=qd.description if qd.description is not None else "",
+            )
+            for qd in self._queries.values()
+        ]
+        query_definitions.sort(key=lambda qd: qd.name)
+        signal_definitions = [
+            temporalio.api.sdk.v1.WorkflowInteractionDefinition(
+                name=sd.name if sd.name is not None else "",
+                description=sd.description if sd.description is not None else "",
+            )
+            for sd in self._signals.values()
+        ]
+        signal_definitions.sort(key=lambda sd: sd.name)
+        update_definitions = [
+            temporalio.api.sdk.v1.WorkflowInteractionDefinition(
+                name=ud.name if ud.name is not None else "",
+                description=ud.description if ud.description is not None else "",
+            )
+            for ud in self._updates.values()
+        ]
+        update_definitions.sort(key=lambda ud: ud.name)
+        wf_def = temporalio.api.sdk.v1.WorkflowDefinition(
+            type=self._info.workflow_type,
+            query_definitions=query_definitions,
+            signal_definitions=signal_definitions,
+            update_definitions=update_definitions,
+        )
+        cur_details = self.workflow_get_current_details()
+        return temporalio.api.sdk.v1.WorkflowMetadata(
+            definition=wf_def, current_details=cur_details
+        )
+
+    def _timer_impl(
+        self,
+        delay: float,
+        callback: Callable[..., Any],
+        *args: Any,
+        context: Optional[contextvars.Context] = None,
+        options: Optional[_TimerOptions] = None,
```

Comment:
Pedantic, but I think it reads better if require this `options` kwarg and move the `_next_timer_options` extraction stuff to `call_later`

Reply:
Works for me